test/e2e_env/kubernetes/meshmetric/meshmetric.go

```diff
@@ -0,0 +1,249 @@
+		Expect(err).ToNot(HaveOccurred())
+
+		// then
+		Eventually(func(g Gomega) {
+			stdout, _, err := client.CollectResponse(
+				kubernetes.Cluster, "test-server-0", "http://"+net.JoinHostPort(podIp, "8080")+"/metrics",
+				client.FromKubernetesPod(namespace, "test-server-0"),
+			)
+
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(stdout).ToNot(BeNil())
+			// metric from envoy
+			g.Expect(stdout).To(ContainSubstring("envoy_http_downstream_rq_xx"))
+		}).Should(Succeed())
+	})
+
+	It("Default MADS server response", func() {
+		// given
+		Expect(kubernetes.Cluster.Install(BasicMeshMetricForMesh("main-mesh-policy", mainMesh))).To(Succeed())
+		Expect(kubernetes.Cluster.Install(BasicMeshMetricForMesh("secondary-mesh-policy", secondaryMesh))).To(Succeed())
+
+		// then
+		Eventually(func(g Gomega) {
+			assignment, err := kubernetes.Cluster.GetKuma().GetMonitoringAssignment(mainPrometheusId)
+			g.Expect(err).ToNot(HaveOccurred())
+
+			madsResponse := MonitoringAssignmentResponse{}
+			g.Expect(json.Unmarshal([]byte(assignment), &madsResponse)).To(Succeed())
+			// all DPPs from both meshes in single MADS response
+			g.Expect(madsResponse.Resources).To(HaveLen(6))
+		}).Should(Succeed())
+
+		// and same response for secondary backend
+		Eventually(func(g Gomega) {
+			assignment, err := kubernetes.Cluster.GetKuma().GetMonitoringAssignment(secondaryPrometheusId)
+			g.Expect(err).ToNot(HaveOccurred())
+
+			madsResponse := MonitoringAssignmentResponse{}
+			g.Expect(json.Unmarshal([]byte(assignment), &madsResponse)).To(Succeed())
+			// all DPPs from both meshes in single MADS response
+			g.Expect(madsResponse.Resources).To(HaveLen(6))
+		}).Should(Succeed())
+	})
+
+	It("MADS server response for specific Prometheus backend", func() {
+		// given
+		Expect(kubernetes.Cluster.Install(MeshMetricWithSpecificPrometheusBackend("main-mesh-policy", mainMesh, mainPrometheusId))).To(Succeed())
+		Expect(kubernetes.Cluster.Install(MeshMetricWithSpecificPrometheusBackend("secondary-mesh-policy", secondaryMesh, secondaryPrometheusId))).To(Succeed())
+
+		// then
+		Eventually(func(g Gomega) {
+			assignment, err := kubernetes.Cluster.GetKuma().GetMonitoringAssignment(mainPrometheusId)
+			g.Expect(err).ToNot(HaveOccurred())
+
+			madsResponse := MonitoringAssignmentResponse{}
+			g.Expect(json.Unmarshal([]byte(assignment), &madsResponse)).To(Succeed())
+			// all DPPs from primaryMesh for primary Prometheus backend
+			g.Expect(madsResponse.Resources).To(HaveLen(3))
+		}).Should(Succeed())
+
+		// and
+		Eventually(func(g Gomega) {
+			assignment, err := kubernetes.Cluster.GetKuma().GetMonitoringAssignment(secondaryPrometheusId)
+			g.Expect(err).ToNot(HaveOccurred())
+
+			madsResponse := MonitoringAssignmentResponse{}
+			g.Expect(json.Unmarshal([]byte(assignment), &madsResponse)).To(Succeed())
+			// all DPPs from secondaryMesh for secondary Prometheus backend
+			g.Expect(madsResponse.Resources).To(HaveLen(3))
```

Comment:
can we then make this 2 or 4 elements? if we only check length then this could be mixed up

Reply:
changed assertion to more precise